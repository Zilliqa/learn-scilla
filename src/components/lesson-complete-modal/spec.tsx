import React from 'react';
import { shallow } from 'enzyme';
import * as renderer from 'react-test-renderer';

import LessonCompleteModal from '.';
const t = (s) => s;
const submit = () => console.log('submit');
const closeModal = () => console.log('close modal');

describe('Cheat Sheet Modal tests', () => {
  const baseComponent = () => (
    <LessonCompleteModal t={t} isModalVisible={false} onSubmit={submit} closeModal={closeModal} />
  );

  describe('basic tests', () => {
    it('matches the snapshot', () => {
      const tree = renderer.create(baseComponent()).toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('renders the component', () => {
      const wrapper = shallow(baseComponent());
      const assertion = wrapper.find('[data-test-id="lesson-complete-modal"]').length;
      expect(assertion).toBe(1);
    });
  });
});
