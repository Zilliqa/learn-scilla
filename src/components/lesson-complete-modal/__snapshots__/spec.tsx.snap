// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cheat Sheet Modal tests basic tests matches the snapshot 1`] = `
<div
  data-test-id="lesson-complete-modal"
/>
`;
